hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutFromLoadAllOperation.java

```diff
@@ -70,8 +71,12 @@ public void run() throws Exception {
             Object previousValue = recordStore.putFromLoad(key, value);
 
             callAfterPutInterceptors(value);
+            // Null-pointer here, in case of null key loaded by MapLoader
+            checkNotNull(key, "Key loaded by a MapLoader cannot be null.");
```

Comment:
@ahmetmircik I didn't want to change the existing logic, though it's slightly broken. 
The LoadAllOperation is called here, the mapper will map null keys to partitions anyway.
https://github.com/tombujok/hazelcast/blob/16bd03c7794322b5989d1bd5641c80352c1117cb/hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java#L342

If I add validation before, the batch won't be half-loaded - since now it loads the data until it reaches null, I didn't want to change that... Will need to be fixed in 4.0